experimental/web_dashboard/lib/src/api/mock.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:uuid/uuid.dart' as uuid;
+
+import 'api.dart';
+
+class MockDashboardApi implements DashboardApi {
+  @override
+  final EntryApi entries = MockEntryApi();
+
+  @override
+  final ItemApi items = MockItemApi();
+}
+
+class MockItemApi implements ItemApi {
+  Map<String, Item> _storage = {};
+  StreamController<List<Item>> _streamController =
+      StreamController<List<Item>>.broadcast();
+
+  @override
+  Future<Item> delete(String id) async {
+    _broadcast();
+    return _storage.remove(id);
+  }
+
+  @override
+  Future<Item> get(String id) async {
+    return _storage[id];
+  }
+
+  @override
+  Future<Item> insert(Item item) async {
+    var id = uuid.Uuid().v4();
+    var newItem = Item(item.name)..id = id;
+    _storage[id] = newItem;
+    _broadcast();
+    return newItem;
+  }
+
+  @override
+  Future<List<Item>> list() async {
+    return _storage.values.toList();
+  }
+
+  @override
+  Future<Item> update(Item item, String id) async {
+    _storage[id] = item;
+    return item..id = id;
+  }
+
+  Stream<List<Item>> subscribe() {
```

Comment:
IIRC, in other projects I'm typically see something like:

```
Stream<List<Item>> get itemStream => _streamController.stream;
```

Is this a pattern you've seen elsewhere?

Reply:
(gentle ping)

Reply:
fixed